### PR TITLE
Spinnaker v0.2.0

### PR DESCRIPTION
**Pull request by viglesiasce**
1. Update Spinnaker components to their latest versions
1. Pin Minio version and use latest dependent chart
1. Simplify access pattern to only need to expose 1 service (Deck)
1. Enable using GCR for images within Spinanker
1. Automatically update components on config change
1. Add support for storing configuration in Google Cloud Storage

### CODE REVIEW

**Comment by viglesiasce** [comment]
cc @ethanfrogers

**Comment by viglesiasce** [comment]
cc @lachie83

File: stable/spinnaker/templates/configmap/spinnaker-config.yaml [hunk 1]
```diff
@@ -6,7 +6,107 @@ metadata:
     app: {{ template "fullname" . }}
 data:
   clouddriver.yml: |
-{{.Files.Get "config/clouddriver.yml" | indent 4}}
+    server:
```
Comment (lachie83) [inline stable/spinnaker/templates/configmap/spinnaker-config.yaml:9]: What's the reasoning behind mixed mode Files.Get and having the file embedded in the configMap?

File: stable/spinnaker/templates/deployments/deck.yaml [hunk 2]
```diff
@@ -6,22 +6,24 @@ metadata:
     app: {{ template "fullname" . }}
     component: deck
 spec:
+  strategy:
+    rollingUpdate:
+      maxUnavailable: 0
   replicas: 1
   template:
     metadata:
+      annotations:
```
Comment (lachie83) [inline stable/spinnaker/templates/deployments/deck.yaml:15]: 👍

File: stable/spinnaker/templates/configmap/spinnaker-config.yaml (hunk 1 above)
Comment (viglesiasce) [reply stable/spinnaker/templates/configmap/spinnaker-config.yaml:9]: Some of the files do not need to be changed based on the config, some do..

**Comment by viglesiasce** [comment]
@k8s-bot e2e test this

**Comment by viglesiasce** [comment]
@k8s-bot e2e test this

**Comment by viglesiasce** [comment]
@k8s-bot e2e test this

**Comment by lachie83** [comment]
@viglesiasce the chart installs but I'm unable to create an application or a pipeline.

**Comment by viglesiasce** [comment]
@lachie83 is minio working as expected? Usually those issues are due to not being able to use minio/gcs

**Comment by lachie83** [comment]
Tried again today and it worked just fine. The only thing I did yesterday was disable persistence storage in the minio dependency

**Comment by lachie83** [comment]
LGTM

**Comment by desaiak** [comment]
@viglesiasce The chart installs fine but unable to create new application or project. k8s version 1.5.5

 kubectl --namespace myspin get pods
NAME                                          READY     STATUS    RESTARTS   AGE
spin-minio-3073839406-hfszt                   1/1       Running   0          11m
spin-redis-3407887992-pp21f                   1/1       Running   0          11m
spin-spinnaker-clouddriver-1566332209-ccv00   1/1       Running   0          11m
spin-spinnaker-deck-4059313393-pf2l3          1/1       Running   0          11m
spin-spinnaker-echo-1040739406-h69d9          1/1       Running   0          11m
spin-spinnaker-front50-3871954802-c8tj0       1/1       Running   0          11m
spin-spinnaker-gate-1521437698-hw5fd          1/1       Running   1          11m
spin-spinnaker-igor-2331563276-tx34h          2/2       Running   0          11m
spin-spinnaker-orca-655780967-8mmbk           1/1       Running   0          11m
spin-spinnaker-rosco-2899501594-9btnf         1/1       Running   0          11m

**Comment by FengyunPan** [comment]
Hi, everyone, what can I do when I get the following err:
# helm install --name spinnaker stable/spinnaker --debug 
Created tunnel using local port: '39865'
SERVER: "localhost:39865"
Fetched stable/spinnaker to spinnaker-0.2.0.tgz
CHART PATH: /spinnaker-0.2.0.tgz
Error: parse error in "spinnaker/templates/configmap/spinnaker-config.yaml": template: spinnaker/templates/configmap/spinnaker-config.yaml:432: function "first" not defined

**Pull request merged**